src/components/GlobalPreferences/GlobalPreferences.js

```diff
@@ -0,0 +1,208 @@
+  const handleNavigation = useCallback(
+    index => {
+      onScreenChange(PATHS[index])
+    },
+    [onScreenChange]
+  )
+
+  useEffect(() => {
+    if (!path) {
+      setSectionIndex(null)
+      return
+    }
+    const index = PATHS.findIndex(item => path.startsWith(item))
+
+    if (index !== NETWORK_INDEX) {
+      return
+    }
+    setSectionIndex(index === -1 ? null : index)
+
+    // subsection is the part after the PATH, e.g. for `?p=/notifications/verify` - `/verify`
+    const subsection = index !== -1 ? path.substring(PATHS[index].length) : null
+
+    setSubsection(subsection)
+    // Does the current path start with any of the declared route paths
+  }, [path, sectionIndex])
+
+  return { sectionIndex, subsection, handleNavigation }
+}
+
+function Close({ compact, onClick }) {
+  const theme = useTheme()
+  return (
+    <div
+      css={`
+        position: absolute;
+        right: 0;
+        padding-top: ${2.5 * GU}px;
+        padding-right: ${3 * GU}px;
+
+        ${compact &&
+          `
+            padding-top: ${2 * GU}px;
+            padding-right: ${1.5 * GU}px;
+          `}
+      `}
+    >
+      <ButtonIcon onClick={onClick} label="Close">
+        <IconClose
+          css={`
+            color: ${theme.surfaceIcon};
+          `}
+        />
+      </ButtonIcon>
+    </div>
+  )
+}
+
+function AnimatedGlobalPreferences(props) {
+  const { sectionIndex, handleNavigation } = useGlobalPreferences({
+    path: props.path,
+    onScreenChange: props.onScreenChange,
+  })
+
+  const { below } = useViewport()
+  const compact = below('medium')
+  const theme = useTheme()
+
+  return (
+    <Transition
+      native
+      items={sectionIndex !== null}
+      from={{ opacity: 0, enterProgress: 0, blocking: false }}
+      enter={{ opacity: 1, enterProgress: 1, blocking: true }}
+      leave={{ opacity: 0, enterProgress: 1, blocking: false }}
+      config={springs.smooth}
+    >
+      {show =>
+        show &&
+        /* eslint-disable react/prop-types */
+        // z-index 2 on mobile keeps the menu above this preferences modal
+        (({ opacity, enterProgress, blocking }) => (
+          <AnimatedDiv
+            style={{
+              zIndex: 1,
+              pointerEvents: blocking ? 'auto' : 'none',
+              opacity,
+              transform: enterProgress.interpolate(
+                v => `
+                  translate3d(0, ${(1 - v) * 10}px, 0)
+                  scale3d(${1 - (1 - v) * 0.03}, ${1 - (1 - v) * 0.03}, 1)
+                `
+              ),
+            }}
+            css={`
+              position: fixed;
+              top: 0;
+              bottom: 0;
+              left: 0;
+              right: 0;
+              overflow: auto;
+              min-width: 360px;
+              padding-bottom: ${2 * GU}px;
+              border-top: 2px solid ${theme.accent};
+              background: ${theme.surface};
+              ${breakpoint('medium', `padding-bottom:0;`)}
```

Comment:
Ah... yes... this the GlobalPreferences is probably the last bit in the client that hasn't been cleaned up with the design system.

We should use `useViewport()` instead of `breakpoint()`! (And please also update the client when doing so!)